.github/workflows/build.yml

```
@@ -33,28 +33,34 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v4
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
+    - name: Build tfx-bsl
+      id: build-tfx-bsl
+      uses: ./.github/reusable-build
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Upgrade pip
-      run: |
-        python -m pip install --upgrade pip
 
-    - name: Build the manylinux2010 image
-      run: docker compose build manylinux2010
-
-    - name: Build the package for Python ${{ matrix.python-version }}
-      run: |
-        version="${{ matrix.python-version }}"
-        docker compose run -e PYTHON_VERSION=$(echo "$version" | sed 's/\.//') manylinux2010
-
-    - name: Upload wheel artifact for Python ${{ matrix.python-version }}
-      uses: actions/upload-artifact@v3
-      with:
-        name: tfx-bsl-wheel-py${{ matrix.python-version }}
-        path: dist/*.whl
+  upload_to_pypi:
+    name: Upload to PyPI
+    runs-on: ubuntu-latest
+    if: (github.event_name == 'release' && startsWith(github.ref, 'refs/tags')) || (github.event_name == 'workflow_dispatch')
```

```suggestion
    if: (github.event_name == 'release' && startsWith(github.ref, 'refs/tags'))
```

This job already doesn't trigger on workflow dispatch, so there's no need for this here.

I think the triggers at the top of this workflow should look like this:

```yaml
on:
  pull_request:
  release:
    types: [published]
```

because currently this workflow won't trigger on releases. Sorry I missed this in the previous PR!

---

Yep makes sense. Thanks for pointing it out. We would also need to trigger on `push` to main to be able to run on merges to main, right?

---

Sure, especially if we have no branch protections and people will be pushing directly to `master` :neutral_face: